org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -60,6 +82,102 @@ public void start(BundleContext bundleContext) {
 		initializeJDTOptions();
 		projectsManager = new ProjectsManager(preferenceManager);
 		logInfo(getClass()+" is started");
+		configureProxy();
+	}
+
+	private void configureProxy() {
```

Comment:
I'd like to see a test for configureProxy() which shows http* SysProps are preserved after calling configureProxy(), if possible